addons/binding/org.openhab.binding.vera2/src/main/java/org/openhab/binding/vera2/internal/converter/VeraDeviceStateConverter.java

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.vera2.internal.converter;
+
+import static org.openhab.binding.vera2.VeraBindingConstants.BATTERY_CHANNEL;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.vera2.controller.json.Device;
+import org.slf4j.Logger;
+
+/**
+ * The {@link VeraDeviceStateConverter} is responsible for converting Vera device level to openHAB states
+ *
+ * @author Dmitriy Ponomarev
+ */
+public class VeraDeviceStateConverter {
+    public static State toState(Device device, Channel channel, Logger logger) {
+        if (channel.getUID().getId().split("-")[0].equals(BATTERY_CHANNEL)) {
+            return getMultilevelState(device.batterylevel);
+        }
+
+        int category = Integer.parseInt(device.category);
+        int subcategory = Integer.parseInt(device.subcategory);
+        switch (category) {
+            case 0:
+            case 1: // Interface
+                break;
+            case 2: // Dimmable Light
+                switch (subcategory) {
+                    case 1:
+                    case 2:
+                    case 3:
+                        return getPercentState(device.level);
+                    case 4:
+                        return getPercentState(device.level);// TODO getColorState(device.light);

```

Comment:
There is a TODO, what should be done and can it be completed before merging the PR?